calcom/cal.com
# feat: attribute weights and virtual queues

## What does this PR do?

Allows to add weights to hosts' attribute values. If the attribute has weights enabled and routing was rerouted with attributed matching the attribute weights will be used instead of the event type weight. When attribute weight is used then also virtual queues will be used. A virtual queues is defined by the `chosenRouteId` and the first attribute that uses 'Value of field..'

Limitations (will be added later): 
- No UI for adding weights to attributes
- No UI for choosing the attributeField we want to use for weights and virtual queue creation
- Only one attribute can have weights enabled

- Fixes CAL-4655
- Fixes CAL-4549

## How should this be tested?

- Set isWeightsEnabled to true for one attribute
- give team members attributes 
- create a routing form with several routes and use 'Value of field ...' for the attribute where you enabled weight
- For RR it should only consider bookings from that queue with the same attribute values (of the attribute where weights are enabled)
- Weights from RR event type will be ignored and only the attributes weights are considered
